### DISCUSSION TITLE
How to get grain size distribution through the image generated from PoreSpy

### DISCUSSION THREAD

**Discussion by morendong** [PMEAL/porespy#883] [Q&A] [answered]
I have generated a porous media cube through PoreSpy.
and the pore size distribution can be obtained, it is very nice!
But, I want to get information on soil texture, such as grain size distribution.
Can you give me some advice on how to deal with it?

**Comment by jgostick** [comment, marked as answer]
Firstly, this is a Discussion, not an issue/bug report, so I will move it there.  Secondly, this is an easy answer...just invert the image, like `solids = ~voids`, or `solids = voids == 0)`

**Reply by morendong** [reply]
Thanks for your kind reply.
I plan generate soil by PoreSpy, and do some research about the grain and pore size distribution.  Following your examples, I used polydisperse_spheres generated the soil images, but the actual size is not clear for me. Can you give me advices on how to do it?
In addition, as you mentioned above, invert the image, the idea is very nice, but as a beginner, I don’t know how to make it come true. If you can list the example about it, that would be helpful for me! Thanks again for your and your excellent packages including openpnm and porespy.
